dotbot/adapter.py

```diff
@@ -74,7 +81,9 @@ def _byte_received(byte):
         if self.port == "sailbot-simulator":
             self.serial = SailBotSimulatorSerialInterface(_byte_received)
         elif self.port == "dotbot-simulator":
-            self.serial = DotBotSimulatorSerialInterface(_byte_received)
+            self.serial = DotBotSimulatorSerialInterface(
```

Comment:
I think in the medium term the simulator could become an adapter itself, instead of being a special case of the serial interface.
I understand that the current structure is previous to the current PR, so I am not asking for changes at this point, more for an opinion for what makes sense.